Documentation/guides/EmbeddedDSOs.md

```diff
@@ -0,0 +1,67 @@
+# Embedded DSOs
+
+Android v6.0 (API-23) introduced a new way of dealing with the native
+shared libraries shipped in the `.apk`. Before API-23, the libraries
+would be always extracted and placed in the application data
+directory, thus occupying more space than necessary. API-23 added a
+new `AndroidManifest.xml` attribute,
+`//application/@android:extractNativeLibs`, which if set to `false`
+makes Android *not* extract the libraries to the filesystem.  API-23
+added a way to instead load those libraries directly from the `.apk`.
+In order to support that there are a few requirements which this
+commit implements:
+
+  * DSO (`.so`) files must be *stored uncompressed* in the `.apk`.
+  * `<application android:extractNativeLibs="false"/>` must be set
+  * DSOs in the `.apk` must be aligned on the memory page boundary;
+    `zipalign -p` takes care of this.
+
+This commit also implements `libmonodroid.so` suport for loading our
+DSOs directly from the `.apk`.  This operation mode is enabled by the
+presence of the `$__XA_DSO_IN_APK` environment variable.  This
+environment variable is inserted into the application's environment
+by way of placing it in the environment file (a file part of the
+Xamarin.Android App project that has the `@(AndroidEnvironment)`
+build action).  In this mode, the DSOs are *no longer* looked up in
+the application data directory but only in the override directories
+(if the APK is built in Debug configuration) and the `.apk` itself.
+
+Before proper MSBuild support, in order to activate the above mode,
```

Comment:
This section isn't needed.

d15-9 doesn't have *any* DSO support.  The first release with even rudimentary DSO support is d16-0 P1, which *I* don't care about, and will result in madness. ;-)

All that matters is what d16-0 will contain, and it'll *certainly* contain the new mechanism introduced here.